src/inputs/ObjectSearch.tsx

```diff
@@ -278,19 +284,19 @@ class ObjectSearch extends Component<IObjectSearchProps> {
   }
 
   public render () {
-    const { id, onAddNew } = this.injected
+    const { id, onAddNew, searchOnEmpty } = this.injected
       , isLoading = this.isLoading.isTrue
-      , showNoResultsOption = this.hasSearch && !this.hasOptions
+      , showNoResultsOption = (this.hasSearch || searchOnEmpty) && !isLoading && !this.hasOptions
```

Comment:
This boolean logic is a little complex. Can you split it into multiple variables?

Reply:
Better?
```ts
      , canSearch = this.hasSearch || searchOnEmpty
      , showNoResultsOption = canSearch && !isLoading && !this.hasOptions
```